### ISSUE TITLE
Separate "what SBOMs available" query from "return particular SBOM" query

### ISSUE THREAD

**Issue by sparrell** [oasis-tcs/openc2-ap-sbom#34]
The SBOM AP must handle the following conditions:
- "one" SBOM available in one format in one serialization
- multiple SBOMs available (eg one for application, one for container application runs in)
- multiple formats available (eg CycloneDX, SPDX, SWID)
- multiple serializations available - now including JSON
- multiple versions of application (eg SBOM for before patch and for after patch)
- multiple versions of SBOM (eg SBOM updated with new information even though application itself didn't change)
- SBOMs may be present on device (and hence returnable with another query command) or SBOMs may be elsewhere (eg a URI)

There are more choices than when the AP was first conceived and this is a proposal to change a different approach. The current draft contains a single query that asks for a single sbom qualified by prioritized list of preferred format/serializations (and no JSON serializations were available at that time). 

I propose we separate into two queries: (1) querying what SBOMs available where (2) query to return a specific SBOM in a specific format in a specific serialization.

This should handle all the cases listed above (which the current text does not) as well as be more flexible for future changes. It also simplifies implementation - at least in my particular software implementation

**Comment by dlemire60** [comment]
I agree with all of the conditions @sparrell has described, and I'd like to add one to the list:

- SBOM to be returned to Producer or delivered to Producer-specified destination

In the PACE scenario, if SBOMs are either large or numerous, it's more efficient if they can be sent directly to the Posture Attributes Repository (PAR) than returned to the Producer for the Producer to then simply relay them to the PAR. So the second query described above should include some kind of delivery argument. At a minimum I think there need to be two options:

1. Return to requester (i.e., the sender of the query)
2. Deliver to {uri} using {protocol}

In the PACE example, option 2 might look something like:

`Deliver to {address of PAR} using {PAR GraphQL-based API}`

The OpenC2 response to the second query would be sent after delivery to the specified address has succeeded or failed.

**Comment by sparrell** [comment]
Discussed at 2/23/2022 TC working meeting. Agreement in principle to both Duncan's proposal and Dave's addition. Leaving open for next week for discussion/changes/objections and will discuss again next week and start working actual text for spec based on these suggestions.

**Comment by dlemire60** [comment]
See also relevant discussion associated with [PACE PR 31](https://github.com/opencybersecurityalliance/PACE/pull/31), in particular disputing the approach of sending collected attributes directly from the Component to the PAR.
